### PR TITLE
Add flag to get the latest release from a specified major version

### PR DESCRIPTION
**Pull request by jedkohjk**
Resolves reposense/RepoSense#2208

# Proposed commit message

Add flag to get the latest release from a specified major version

# Other information

Use the flag `--latest` or `-l` and specify the major version after the flag.

Was initially planning to make it as an extension of `--tag` or `t`, but parsing wildcards in the middle of strings may be complicated. It may also mean that we are unable to name versions containing `*` or the specific wildcard character used. Hence, I introduced a new flag instead.

### CODE REVIEW

File: get-reposense.py [hunk 1]
```diff
@@ -16,12 +16,32 @@ def parse_args():
...
+            print('GitHub API has exceed the rate limit.')
+            exit(1)
+        releases = response.json()
+        if not releases:
+            print('Error, the provided tag does not exist!')
+            exit(1)
+        for i in releases:
+            release_tag = i['tag_name']
+            if release_tag[:len(tag)] == tag and \
+                    not (release_tag[len(tag):len(tag)+1].isdigit() and tag[-1].isdigit()):
```
Comment (gok99) [inline get-reposense.py:40]: Nit: some tags look like `v1.9rc` which could match with `v1.9` depending on the order of releases but are not technically official releases - in general, release versions could include non-numeric characters. 

Maybe we can split by `.` and compare each section (or something else that makes sense)

**Review by gok99 — CHANGES_REQUESTED** [review]
Really sorry for the review delays! Introducing the `latest` flag will probably avoid confusion.

File: get-reposense.py (hunk 1 above)
Comment (jedkohjk) [reply get-reposense.py]: Hi, thanks for the review, I've updated it accordingly

**Review by gok99 — APPROVED** [review]

**Comment by jedkohjk** [comment]
> @jedkohjk Did you intend to commit RepoSense.jar and logs? LGTM otherwise!

Oh, I didn't notice. I've removed them. Thank you!

**Review by jonasongg — APPROVED** [review]
LGTM!

**Comment by damithc** [comment]
@jedkohjk Changes in this PR does not mention the use of wild card anywhere. So, the users will not be aware that a wildcard can be used?

**Comment by jedkohjk** [comment]
A new flag is used instead of a wildcard for an existing flag

**Comment by jedkohjk** [comment]
@damithc It is mentioned in `run.sh`, which users will have to edit if they do not just want the most recent release of RepoSense. Of course, it will be good to update the RepoSense user guide too, but that's in a different branch and will require a different pull request. I can do it later today.

**Comment by damithc** [comment]
> @damithc It is mentioned in `run.sh`, which users will have to edit anyway if they do not just want the most recent release of RepoSense. Of course, it will be good to update the RepoSense user guide too, but that's in a different branch and will require a different pull request. I can do it later today.

It says `### ./get-reposense.py --latest v1.6           # Gets the latest version of a specific tag` 
But it's not clear what is meant by `latest version of a specific tag`. 

Perhaps something like the following?
`### ./get-reposense.py --latest v1.6           # Gets the latest version with the given version prefix e.g., v1.6.1`

**Comment by jedkohjk** [comment]
> > @damithc It is mentioned in `run.sh`, which users will have to edit anyway if they do not just want the most recent release of RepoSense. Of course, it will be good to update the RepoSense user guide too, but that's in a different branch and will require a different pull request. I can do it later today.
> 
> It says `### ./get-reposense.py --latest v1.6           # Gets the latest version of a specific tag` 
> But it's not clear what is meant by `latest version of a specific tag`. 
> 
> Perhaps something like the following?
> `### ./get-reposense.py --latest v1.6           # Gets the latest version with the given version prefix e.g., v1.6.1` 
> 

Alright, I'll do it later today (probably tonight) as I do not have access to my laptop at the moment.

**Comment by jedkohjk** [comment]
I've made the following 2 PRs addressing the discussion today:

#13 
reposense/RepoSense#2244

**Pull request merged**